### DISCUSSION TITLE
[keyball46] キーの左右が反転する

### DISCUSSION THREAD

**Discussion by satoshi5454** [Yowkees/keyball#455] [Q&A]
お世話になります。
keyball46 を組み立て終わり、正規のファームウェアを QMK Toolbox を使って書き込みました。
トラックボールを右手側に付けましたので keyball_keyball46_via.hex を左右両方の ProMicro に書き込んでおります。
ここで、トラックボールを付けた右側の ProMicro に USB 接続しますと正しく動作するのですが、
左側の ProMicro に USB 接続しますと、左右のキーが反転してしまいます。
(左側の B キーを押すと M が入力される、のような)
Remap で見る限り、左側の ProMicro に接続している場合も、左右は正しくキーマップされているようなのですが。。。
確認方法や対応方法などご教示いただけますでしょうか。
よろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

製作いただいたのは、【Keyball46】で間違いないでしょうか？おそらく２年ほど前にご購入されたものと思います。
ビルドガイドが少々古くQMK ToolBoxをお勧めしていて恐縮ですが、下記リンクのRemapにて最新のファームウェア（Keyball46_via_v1.2.0）を両手に書き込み直していただけますでしょうか。
https://remap-keys.app/catalog/VeVqQfOJkrn6wYY0IEyy/firmware

また、左右判定ジャンパーは左手側のみに正しく半田付けされていますでしょうか？
半田付け面のお写真をいただければ何かアドバイスできると思います。

お手数ですが、ご確認お願いいたします。

**Reply by satoshi5454** [reply]
早々のご連絡をありがとうございます。
はい、Keyball46 になります。
(購入させていただいたものの、なかなか作成のタイミングがなくずっとしまっていたのですが、2 年も経ってしまっておりました；)

ご提示いただきましたリンク先にて、最新のファームウェアを書き込んだのですが、変わりなしでした。
(正しく書き込めているとは思いますが、何か確認方法はありますでしょうか)

左右判定ジャンパーも、左手側にのみ半田付けしております。
写真を添付いたしますので、ご確認いただけますと幸いです。

![keyball46_左手](https://github.com/Yowkees/keyball/assets/938874/ea6ffff8-cb66-4331-8d31-0c32acbef1b9)
![keyball46_右手](https://github.com/Yowkees/keyball/assets/938874/2ab5ca04-14a6-4951-88c1-202d4295dc07)

よろしくお願いいたします。

**Comment by Yowkees** [comment]
お写真ありがとうございます。
Keyballシリーズのかなり販売初期にご購入いただけたのですね。誠にありがとうございます。

まず、左右判定機能に対応するダイオードが下画像の丸で示したものになります。これがテンプラしていないかご確認ください。
ダイオード左右の端子を半田ごてで触ってみると、テンプラしていれば部品が脱落しますので判ります。

次にTRSソケットの半田付け（同じく丸で囲った部分）の半田が少ないようです。
ここは基板パッドが大きいので熱が伝わりにくいです。半田ごてで十分に温めて半田を追加して、半田がパッドに馴染んで山型になるようにしてください。右手側も同様にTRSコネクタに半田を足して下さい。

ひとまず以上を試していただければ幸いです。ご対応お願いいたします。
![KB46_Leftfix](https://github.com/Yowkees/keyball/assets/84209772/25dbcd3d-503c-4862-b5d2-5c7ae5b18911)

**Comment by satoshi5454** [comment]
早々のご確認をありがとうございます。
半田付けの確認と追加、承知いたしました。

子供が休みに入り、半田を触れる時間の確保が難しくなってしまい、ご連絡まで少し時間がかかってしまいそうです。
申し訳ありませんが、引き続き、どうぞよろしくお願いいたします。

**Comment by satoshi5454** [comment]
ご連絡が遅くなってしまい、申し訳ありません。
ご指摘いただきました、左右判定ジャンパーと TRS ソケットの半田を足しましたところ、左右のキーが反転する症状は治りました！
ありがとうございます。

ただ、今度は、左側の ProMicro に USB を挿しますと、右手側 N キーが反応しなくなりました。
(右側の ProMicro に USB を挿している場合には反応しますので、N キーの問題ではなさそうです)
また、左右どちらの ProMicro に USB を挿しましても、トラックボールが反応いたしません。
(今回の半田対応前には反応しておりました)

現在の半田付け面の写真を添付いたします。
![keyball46_left](https://github.com/Yowkees/keyball/assets/938874/b1f5b8c0-2806-40f3-b798-af173fac3eb0)
![keyball46_right](https://github.com/Yowkees/keyball/assets/938874/9a1d5479-0bd8-4f2a-b308-66eff6214129)

お手数をおかけいたしますが、確認方法や対応など、ご教示いただけますでしょうか。
よろしくお願いいたします。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

あららら。トラックボールの調子が悪くなってしまいましたか。
右手側Nキーが入力できないのは、トラックボール基板と通信できないために左手側配列と判定されているためです。
左手側にはNキー部分にキーが無いので、このような動作となります。

ということで、トラックボール基板とL字7Pコンスルー、トラックボール近くのジャンパーの半田付けを見直してください。一度、トラックボール基板を取り外して付け直すだけでも改善するかも知れません。
修理前は動いていたようですので、修理中に触った個所をメインに見直していただければ幸いです。
トラックボールが動作すればNキーも入力できるようになるはずです。

以上、お手数ですがご対応お願いいたします。

**Comment by satoshi5454** [comment]
丁寧なご回答をありがとうございます。
症状につきまして、理解できました。

トラックボール周りの確認とのこと、承知いたしました。
順番に確認してみます。

またご連絡させていただきます。
引き続き、どうぞよろしくお願いいたします。

**Comment by satoshi5454** [comment]
無事、正常動作を確認できましたので、ご報告いたします。
結論としまして、大きな原因は、メイン基板とボール基板とに隙間ができてしまっていることのようでした。

ご指摘に沿って、トラックボール基板と L 字 7P コンスルーの半田を見直し、少なめの箇所には半田を足し、
ボトム プレートは外した状態で正常動作を確認したのですが、
ボトム プレートを付けると上手く動かないが再発してしまい、改めて見直したところ、
ネジ止めの際に、メイン基板とボール基板とに隙間が生じてしまったことが原因のようでした。
隙間ができないようにネジ止めをし直したところ、正常動作が確認できました。
ビルド ガイドにも丁寧に書いてくださっているにも関わらず、見落としてしまっており、申し訳ありませんでした。

ご丁寧に対応いただき、ありがとうございました。

**Comment by Yowkees** [comment]
ご対応ありがとうございます。ご自身で工夫していただき、無事に動作するようになったようで安心しました。
ぜひ末永くご愛用いただければ幸いです。
